nexus/src/app/sagas/instance_create.rs

```diff
@@ -977,20 +977,11 @@ async fn sic_delete_instance_record(
         &sagactx,
         &params.serialized_authn,
     );
+
     let instance_id = sagactx.lookup::<InstanceUuid>("instance_id")?;
-    let instance_name = sagactx
-        .lookup::<db::model::Instance>("instance_record")?
-        .name()
-        .clone()
-        .into();
-
-    // We currently only support deleting an instance if it is stopped or
-    // failed, so update the state accordingly to allow deletion.
-    // TODO-correctness TODO-security It's not correct to re-resolve the
-    // instance name now.  See oxidecomputer/omicron#1536.
+
     let result = LookupPath::new(&opctx, &datastore)
-        .project_id(params.project_id)
-        .instance_name(&instance_name)
+        .instance_id(instance_id.into_untyped_uuid())
         .fetch()
         .await;
 
```

Comment:
After your change, line 988 has a comment saying "as mentioned in the comment above, we should not be doing lookup by name" - but I think you're removing that comment.

Maybe just update this to include the bit about idempotency, but drop the bit about the name lookup?

Reply:
Good catch, fixed that in 0a7601bbaff13944f14d475f96e211b8af42d9b7!